src/context/viewer-context.tsx

```diff
@@ -56,6 +56,18 @@ function useAuthedViewer() {
     const accessToken = get(queryHash, 'access_token')
     const noAccessTokenFound = isEmpty(accessToken)
     const viewerIsPresent = !isEmpty(viewerId)
+    const authToken = getAccessTokenFromCookie()
+
+    if (loggingOut) {
+      const doLogout = async () => {
+        await auth.logout()
+        if (typeof window !== 'undefined') {
+          window.location.href = '/'
+        }
+      }
+      doLogout()
+      return
```

Comment:
How does this return impact the unmounting of the useEffect? I guess this is overriding the bottom of the effect where it:

```javascript
return clearUserMonitorInterval
```

Reply:
it won't be an issue because the interval is never created to be clear since we bail early on this run. Any previous instances will be cleared when its component instance unmounts and before this effect is executed.